esp-hal-common/src/interrupt/riscv.rs

```diff
@@ -565,6 +566,36 @@ mod vectored {
 /// # Safety
 ///
 /// This function is called from an assembly trap handler.
+/// # Safety
+///
+/// This function is called from an assembly trap handler.
+#[cfg(not(esp32c6))]
+#[doc(hidden)]
+#[export_name = "set_prio"]
+unsafe fn handle_priority()->u32{
+    let interrupt_id:usize = mcause::read().bits() & 0x0FFFFFFF; //MSB is whether its exception or interrupt.
+    let intr = &*crate::peripherals::INTERRUPT_CORE0::PTR;
+    let interrupt_priority =  intr.cpu_int_pri_0.as_ptr().offset(interrupt_id as isize).read_volatile();
+    let prev_interrupt_priority = intr.cpu_int_thresh.read().bits();
+    intr.cpu_int_thresh.write(|w| w.bits(interrupt_priority + 1)); //set the prio threshold to 1 more than current interrupt prio
+    unsafe{
+        riscv::interrupt::enable();
+    }
+    prev_interrupt_priority
+    //panic!();
+        
+
+}
+#[cfg(not(esp32c6))]
+#[doc(hidden)]
+#[export_name = "restore_prio"]
+unsafe fn restore_priority(stored_prio:u32){
+    let intr = &*crate::peripherals::INTERRUPT_CORE0::PTR;
+    intr.cpu_int_thresh.write(|w| w.bits(stored_prio)); //set the prio threshold to 1 more than current interrupt prio
```

Comment:
It's important to ensure interrupts are disabled before returning here, I think (technically, I believe any time before [`_start_trap` pops clobber-able machine state off the stack frame](https://github.com/rustbox/esp-riscv-rt/blob/a5a249eaba05da482743112a26a4558d9209b200/src/lib.rs#L425-L429) would work). Otherwise, an ill-timed IRQ can lead to some very spicy results.

```suggestion
    let intr = &*crate::peripherals::INTERRUPT_CORE0::PTR;
    unsafe{
        riscv::interrupt::disable();
    }
    intr.cpu_int_thresh.write(|w| w.bits(stored_prio)); // restores the previous prio threshold 
```

Reply:
Yeah, i agree, i'll get this fixed and rebase this branch to the current implementation, sorry for the confusion and thanks for having a look :D